gp-maven-plugin/src/main/java/com/ibm/g11n/pipeline/maven/GPDownloadMojo.java

```diff
@@ -232,11 +202,18 @@ private void exportLanguageResource(ServiceClient client, BundleFile bf, String
         }
     }
 
-    private String getLanguageId(String languageTag) {
-        String languageId = languageTag;
-        switch (languageIdSeparator) {
+    private String getLanguageId(String gpLanguageTag, LanguageIdSeparator langSep,
+            Map<String, String> langMap) {
+        String languageId = gpLanguageTag;
+        if (langMap != null) {
+            String mappedId = langMap.get(gpLanguageTag);
+            if (mappedId != null) {
+                languageId = mappedId;
+            }
+        }
+        switch (langSep) {
         case UNDERSCORE:
-            languageId = languageTag.replace('-', '_');
+            languageId = languageId.replace('-', '_');
```

Comment:
The input is in BCP47 language tag used by Globalization Pipeline for language identification.  The output - languageId - is for file/directory name or file name prefix/suffix used by some resource bundle implementations. java.util.Locale has a side effect in some cases, and not sure why we want to use ICU4J to just replacing '-' with '_' here.